### PR TITLE
PLFM-7068: Lambda artifact

### PR DESCRIPTION
**Pull request by xschildw**
This PR builds and deploys the lambda artifact used to create a lambda function called by the back-end to convert markdown to htm in Synapse messages.

### CODE REVIEW

File: .github/workflows/deploy.yaml [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+        zip -r ../markdown-it-${{ env.TAG_NAME }}.zip .
+
+    - name: Setup JF
+      uses: jfrog/setup-jfrog-cli@v4
+      env:
+        JF_USER: ${{ secrets.PLATFORM_ARTIFACTORY_USER }}
+        JF_PASSWORD: ${{ secrets.PLATFORM_ARTIFACTORY_PWD }}
+
+    -  run: |
+        echo Deploying artifact ../markdown-it-${{ env.TAG_NAME }}.zip
```
Comment (xschildw) [inline .github/workflows/deploy.yaml]: Just testing connectivity now.

File: md-to-html/core.mjs [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+  const resultHtml = md.render(markdown);
+  console.log("resultHtml:" + resultHtml);
+  let endResult = "";
+
+  if (outputType == 'html') {
+    endResult = resultHtml;
+  } else if (outputType == 'plain') {
+      endResult = htmlToText(resultHtml, { wordwrap: 130 });
+  }
```
Comment (xschildw) [inline md-to-html/core.mjs:55]: We used to return a large .css string here when not (html or plain), now I'm checking in the caller and returning a 400.

File: md-to-html/package.json [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+  ],
+  "dependencies": {
+    "body-parser": "^1.19.0",
+    "cookie-parser": "^1.4.5",
+    "debug": "^4.3.4",
+    "eslint": "^7.32.0",
+    "eslint-plugin-nodeca": "^1.0.0",
+    "html-to-text": "^8.0.0",
+    "juice": "^7.0.0",
+    "markdown-it": "^12.3.0",
```
Comment (jay-hodgson) [inline md-to-html/package.json]: Consider updating the markdown-it dependency to "^14.1.0", and removing markdown-it-emphasis-alt (unnecessary with latest markdown-it release).  
There may be other dependencies that also should be updated

File: md-to-html/core.mjs [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+  const md = markdownIt()
+  md.use(synapsePlugin, '', markdown.baseURL)     
+    .use(markdownItSub)
+    .use(markdownItSup)
+    .use(markdownItCenterText)
+    .use(markdownItSynapseHeading)
+    .use(markdownItSynapseTable)
+    .use(markdownItStrikethroughAlt)
+    .use(markdownItContainer)
+    .use(markdownItEmphasisAlt)
```
Comment (jay-hodgson) [inline md-to-html/core.mjs]: You could remove emphasis-alt if you update markdown-it

File: md-to-html/core.mjs [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+import { htmlToText } from 'html-to-text';
+import juice from 'juice';
+import markdownItSub from 'markdown-it-sub';
+import markdownItSup from 'markdown-it-sup';
+import markdownItCenterText from 'markdown-it-center-text';
+import markdownItSynapseHeading from 'markdown-it-synapse-heading';
+import markdownItSynapseTable from 'markdown-it-synapse-table';
+import markdownItStrikethroughAlt from 'markdown-it-strikethrough-alt';
+import markdownItContainer from 'markdown-it-container';
+import markdownItEmphasisAlt from 'markdown-it-emphasis-alt';
```
Comment (jay-hodgson) [inline md-to-html/core.mjs]: You could remove emphasis-alt if you update markdown-it

**Review by jay-hodgson — APPROVED** [review]

**Pull request merged**